sockeye/model.py

```diff
@@ -298,13 +302,21 @@ def decode_step(self,
             decode_step_module = _DecodeStep(self.embedding_target,
                                                 self.decoder,
                                                 self.output_layer,
-                                                self.factor_output_layers)
+                                                self.factor_output_layers,
+                                                self.knn)
             self.traced_decode_step = pt.jit.trace(decode_step_module, decode_step_inputs)
         # the traced module returns a flat list of tensors
         decode_step_outputs = self.traced_decode_step(*decode_step_inputs)
-        step_output, *target_factor_outputs = decode_step_outputs[:self.num_target_factors]
-        new_states = decode_step_outputs[self.num_target_factors:]
-        return step_output, new_states, target_factor_outputs
+        # +1 for the decoder output, which will be used to generate kNN output
+        step_output, decoder_out, *target_factor_outputs = decode_step_outputs[:self.num_target_factors+1]
+
+        # do the query here because it cannot be traced (jit.ignore does not play well with tracing)
+        knn_output = None
```

Comment:
these 3 lines can be turned into one:
`knn_output = self.knn(decoder_out) if self.knn is not None else None`